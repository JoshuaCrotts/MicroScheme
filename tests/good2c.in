
// Addition of chars should produce a char - no casting. But assigning to an
// int involves casting.

void main() {
    char ca = 1;
    char cb = 2;
    int ia = 1;
    int ib = 2;
    char cc;
    int  ic;
    
    cc = ca * cb;
    ic = cc;
    printd(ic);
}

