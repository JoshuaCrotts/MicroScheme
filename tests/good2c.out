void SEQ (
  void FNDEF main void () (
    void SEQ (
      void DECL ca (char) = 1,
      void DECL cb (char) = 2,
      void DECL ia (int) = 1,
      void DECL ib (int) = 2,
      void DECL cc (char),
      void DECL ic (int),
      char ASN (
        char ID cc,
        char BINOP('*') (
          char ID ca,
          char ID cb)),
      int ASN (
        int ID ic,
        int CAST (
          char ID cc)),
      void FNCALL printd (
        int ID ic))))
